### DISCUSSION TITLE
InspectTree node is not valid node in EditQuery

### DISCUSSION THREAD

**Discussion by Demianeen** [nvim-treesitter/nvim-treesitter#6755] [Q&A] [answered]
I was trying to add some sort of highlighting for the format specifier in go: 
![CleanShot 2024-06-11 at 07 51 02@2x](https://github.com/nvim-treesitter/nvim-treesitter/assets/51330172/5b9da639-ea27-4c87-8b2c-9b54be2500c4)

In `:InspectTree` I could see that there is a format node that is responsible for it. I then used `:EditQuery` to try to add query, but it is saying that the node that I took from `:InspectTree` is an invalid one. Take a look:

https://github.com/nvim-treesitter/nvim-treesitter/assets/51330172/37c3a91f-3e4c-4f95-8e17-ccd2308c76c3

Is there any difference between InspectTree and EditQuery nodes?

**Comment by clason** [comment]
InspectTree is not a query.

**Reply by clason** [reply]
Oh, and make sure that you are actually looking at nodes for Go, not for any injected languages. (Press `I` to toggle languages in the InspectTree view.)

**Reply by Demianeen** [reply]
Then when do you use InspectTree? I thought it is like Treesitter Playground

To second comment:

![CleanShot 2024-06-11 at 07 59 24@2x](https://github.com/nvim-treesitter/nvim-treesitter/assets/51330172/ce121d72-b186-43e7-acf2-5f7d752cebdb)

It says that this is some printf language. Can I then make a query to it instead of go?

**Reply by clason** [reply, marked as answer]
> Then when do you use InspectTree? I thought it is like Treesitter Playground

It is, but it just shows a tree of parsed nodes. Queries are different (captures and whatnot).

> It says that this is some printf language. Can I then make a query to it instead of go?

Yes, but not in the EditQuery window; that only targets the primary language. `:h treesitter-language-injections`
